src/Compilers/Core/Portable/InternalUtilities/GeneratedCodeUtilities.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+
+namespace Roslyn.Utilities
+{
+    internal static class GeneratedCodeUtilities
+    {
+        private static readonly string[] s_autoGeneratedStrings = new[] { "<autogenerated", "<auto-generated" };
+
+        internal static bool IsGeneratedSymbolWithGeneratedCodeAttribute(
```

Comment:
There is a difference between files generated by the build system and those generated by a tool. Tool generated files might be ones you want to error & fix, because they might be generated once and then become part of the source your own.  Also, this was even being confused by files generated by the compiler itself.